### ISSUE TITLE
Forfeited junior loss re-enters fee accounting via permissionless AccrueFees on mode-1 pools

### ISSUE THREAD

**Issue by v1ktorrr0x** [dcccrypto/percolator-stake#188]
## Summary

When the last junior LP exits during an active insurance loss, the program books the absorbed loss as `realized_junior_loss` and bumps `total_returned`. This creates a bookkeeping gap where `vault_physical_balance > total_pool_value()`. On a mode-1 (trading LP) pool, anyone can then transfer tokens directly into the vault and call the permissionless `AccrueFees` instruction to crystallize that gap — plus any donated amount — into `total_fees_earned`, inflating senior share price. Senior LPs extract value that should be dead.

---

## Affected Code

**Last-junior-exit accounting** (`process_withdraw`):
```rust
// src/processor.rs — last junior exit block
let forfeited = pool.junior_balance()
    .saturating_sub(pool.effective_junior_balance());
let forfeited = forfeited.min(net_loss);
if forfeited > 0 {
    pool.total_returned = pool.total_returned
        .checked_add(forfeited)
        .ok_or(StakeError::Overflow)?;
    pool.set_realized_junior_loss(
        pool.realized_junior_loss()
            .checked_add(forfeited)
            .ok_or(StakeError::Overflow)?,
    );
}
pool.set_junior_balance(0);
```

**Fee accrual delta** (`accrue_fees_inner`):
```rust
// src/processor.rs
let pool_value = pool.total_pool_value().ok_or(StakeError::Overflow)?;
if current_balance > pool_value && pool.total_lp_supply > 0 {
    let fee_delta = current_balance - pool_value;
    pool.total_fees_earned = pool.total_fees_earned
        .checked_add(fee_delta)
        .ok_or(StakeError::Overflow)?;
    // ... distribute_fees to tranches
}
```

**`total_pool_value()` subtracts `realized_junior_loss`** (state.rs), so the gap is structural, not incidental.

**`AccrueFees` is permissionless** — any signer can call it:
```rust
// src/processor.rs
let caller = next_account_info(accounts_iter)?; // signer, permissionless
if !caller.is_signer {
    return Err(ProgramError::MissingRequiredSignature);
}
```

---

## Attack Scenario

```
Initial state (mode-1, tranches enabled):
  junior_balance     = 1000
  senior_balance     = 1000
  total_flushed      = 800
  total_returned     = 0
  effective_junior   = 200   (junior absorbed 800 of loss)

Step 1 — last junior exits:
  withdrawal_amount  = 200   (from effective_junior_balance)
  forfeited          = 600   (junior_balance 1000 - effective 200 - already withdrawn 200)
  total_returned    += 600   → 600
  realized_junior_loss = 600
  junior_balance     = 0
  junior_total_lp    = 0

Now:
  total_pool_value() = deposited - withdrawn - flushed + returned + fees - realized_loss
                     = ... (senior portion unchanged, dead 600 sits in vault)

Step 2 — attacker (senior LP holder) donates 1 token to vault:
  vault_physical_balance = senior_balance + 600 (dead) + 1 (donated)

Step 3 — attacker calls AccrueFees permissionlessly:
  pool_value    = total_pool_value()  // does NOT include the 600 dead tokens
  fee_delta     = current_balance - pool_value  // = 601
  total_fees_earned += 601
  senior share price appreciates by 601

Result: attacker recovers 600 tokens of forfeited junior loss + 1 donated token
        via inflated senior redemption price.
```

---

## Impact

- Senior LPs extract value that was explicitly forfeited by juniors at exit — violates the core protocol guarantee that senior must not windfall from junior-absorbed losses.
- The `#161` last-junior-exit logic (designed to protect senior from future recovery) is bypassed by routing through `accrue_fees_inner` rather than `ReturnInsurance`.
- Attack is permissionless (any signer, zero trust required) and requires only 1 lamport of donated tokens as a catalyst on mode-1 pools.
- Scoped to **mode-1 (trading LP) pools only** — `AccrueFees` is gated on `pool_mode != 1` for mode-0 pools.

**Severity: Medium**

---

## Fix

Track forfeited junior losses in a dedicated `dead_vault_tokens` field and subtract it from `current_balance` before computing `fee_delta` in `accrue_fees_inner`:

```rust
// In accrue_fees_inner:
let effective_balance = current_balance
    .saturating_sub(pool.realized_junior_loss()); // dead tokens are not fee revenue
if effective_balance > pool_value && pool.total_lp_supply > 0 {
    let fee_delta = effective_balance - pool_value;
    // ...
}
```

Alternatively, add a dedicated `dead_vault_tokens: u64` field to `StakePool` incremented at the last-junior-exit and decremented only when those tokens are explicitly written off or reclaimed, to make the accounting explicit and auditable.

---

## References

- `#161` — last-junior-exit fair-recovery logic
- `accrue_fees_inner` — `src/processor.rs`
- `process_withdraw` last-junior block — `src/processor.rs`
- `total_pool_value()` — `src/state.rs`

**Comment by dcccrypto** [comment]
Verified NOT_REAL: the math cancels (+total_returned offset by -realized_junior_loss in total_pool_value), so no dead-token gap is crystallized into fees; the premise (mode-0 flush coexisting with mode-1 AccrueFees) is unreachable. Closing.
